File: 2122-count-special-quadruplets/2122-count-special-quadruplets.java
class Solution {
    public int countQuadruplets(int[] nums) {
        int c=0;
        for(int i=0;i<nums.length-3;i++)
        for(int j=i+1;j<nums.length-2;j++)
        for(int k=j+1;k<nums.length-1;k++)
        for(int p=k+1;p<nums.length;p++)
        if(nums[i]+nums[j]+nums[k] == nums[p])
        c++;
        return c;
        
    }
}

File: 2122-count-special-quadruplets/2122-count-special-quadruplets.c
int countQuadruplets(int* a, int n) {
    int i,j,k,z,c;
    c=0;
    for(i=0;i<n;i++)
        for(j=i+1;j<n;j++)
        for(k=j+1;k<n;k++)
        for(z=k+1;z<n;z++)
        if(a[i]+a[j]+a[k]==a[z])
        c++;
    return c;
}